### PR TITLE
MYFACES-4655 fix resolving of FacesValidators without value

### PR DESCRIPTION
**Pull request by NicolaIsotta**
https://issues.apache.org/jira/browse/MYFACES-4655

### CODE REVIEW

File: impl/src/main/java/org/apache/myfaces/cdi/util/CDIUtils.java [hunk 1]
```diff
@@ -120,6 +135,17 @@ public static <T> T get(BeanManager beanManager, Type type, boolean create, Anno
...
 
+    private static String getBeanName(Bean<?> bean)
+    {
+        String name = bean.getName();
+        if (name != null)
+        {
+            return name;
+        }
+        String className = bean.getBeanClass().getSimpleName();
+        return Character.toLowerCase(className.charAt(0)) + className.substring(1);
```
Comment (melloware) [inline impl/src/main/java/org/apache/myfaces/cdi/util/CDIUtils.java:146]: These two lines are the only ones I am not sure about but @tandraschko will weigh in!

**Comment by melloware** [comment]
@tandraschko does this need to be backported and forward ported once accepted?

**Comment by tandraschko** [comment]
can you explain the change?

**Comment by NicolaIsotta** [comment]
Not needed, I already tried, here's what happens if you don't declare forClass and value on FacesConverter:
```
@FacesConverter must have value, forClass or both. Check annotation @FacesConverter on class: org.primefaces.test.MyCdiConverter
```

File: impl/src/main/java/org/apache/myfaces/cdi/util/CDIUtils.java (hunk 1 above)
Comment (tandraschko) [reply impl/src/main/java/org/apache/myfaces/cdi/util/CDIUtils.java:146]: i also wonder about this one - why is this required?
If bean.getName is null, its not a named bean - why would need it then?

Comment (NicolaIsotta) [reply impl/src/main/java/org/apache/myfaces/cdi/util/CDIUtils.java:146]: TBH I'm not sure too - guess it depends on how the validator is registered in CDI?

What I'm sure is that without these lines the fix doesn't work...

Comment (melloware) [reply impl/src/main/java/org/apache/myfaces/cdi/util/CDIUtils.java:146]: It looks like BalusC does this in his CDIUtils for good reason I guess.  he does...

```java
        String className = bean.getBeanClass().getSimpleName();

        return new StringBuilder()
            .append(Character.toLowerCase(className.charAt(0)))
            .append(className.substring(1))
            .toString();
```

**Comment by tandraschko** [comment]
can you provide a PR for 4.1 and 5.0 too?

**Pull request merged**